game/dota_addons/dota_imba/scripts/vscripts/hero/hero_phantom_lancer.lua

```diff
@@ -0,0 +1,206 @@
+--[[
+Author: Shush
+Date: 13.09.2017
+]]
+
+-- CreateEmptyTalents("phantom_lancer")
+
+--------------------------------------
+--			SPIRIT LANCE            --
+--------------------------------------
+imba_phantom_lancer_spirit_lance = imba_phantom_lancer_spirit_lance or class({})
+LinkLuaModifier("modifier_imba_spirit_lance_slow", "hero/hero_phantom_lancer", LUA_MODIFIER_MOTION_NONE)
+
+function imba_phantom_lancer_spirit_lance:OnAbilityPhaseStart()
+	-- Ability properties
+	local caster = self:GetCaster()
+	local ability = self
+	local target = self:GetCursorTarget()
+
+	-- Ability specials
+	local illusion_lance_cast_range = ability:GetSpecialValueFor("illusion_lance_cast_range")
+
+	-- Find all nearby illusions
+	local units = FindUnitsInRadius(caster:GetTeamNumber(),
+	 									caster:GetAbsOrigin(),
+	 									nil,
+	 									illusion_lance_cast_range,
+	 									DOTA_UNIT_TARGET_TEAM_FRIENDLY,
+	 									DOTA_UNIT_TARGET_HERO,
+	 									DOTA_UNIT_TARGET_FLAG_PLAYER_CONTROLLED,
+	 									FIND_ANY_ORDER,
+	 									false)	
+
+	-- Cycle between found units, only include illusions
+	for _, unit in pairs(units) do
+		if unit:GetPlayerOwnerID() == caster:GetPlayerOwnerID() and unit ~= caster then			
+
+			-- Fake casting animation
+			unit:StartGesture(ACT_DOTA_CAST_ABILITY_1)				
+
+			-- Illusions that are moving/attacking are ignored. Idle illusions fake their Spirit Lance
+			if unit:IsIdle() then
+				unit:FaceTowards(target:GetAbsOrigin())
+			end			
+		end
+	end
+
+	return true
+end
+
+function imba_phantom_lancer_spirit_lance:OnSpellStart()
+	-- Ability properties
+	local caster = self:GetCaster()
+	local ability = self
+	local target = self:GetCursorTarget()
```

Comment:
Is it possible to do this?:

```
local ability = self
local caster = ability:GetCaster()
local target = ability:GetCursorTarget()
```

I think it reads better this way, and as a little bonus the vars are alphabetized.

Reply:
Yes, it's possible, but it doesn't matter much. Just got my kind of form by now.